enamel/api/handlers.py

```diff
@@ -45,6 +53,22 @@ def handle_error(error):
     return response
 
 
+def handle_404(error):
+    """Transform a Flask 404 into our error handling flow."""
+    return handle_error(httpexceptor.HTTP404(error.description))
+
+
+def set_request_id():
+    """A before_request function to set required-id."""
+    flask.g.request_id = uuidutils.generate_uuid()
```

Comment:
Have I mentioned that I can't stand thread-local proxy variables masquerading as global variables? :)